nitransforms/linear.py

```diff
@@ -187,8 +187,270 @@ def from_filename(cls, filename, fmt='X5',
+        """Enable using len()."""
+        return len(self._matrix)
+
+    def map(self, x, inverse=False):
+        r"""
+        Apply :math:`y = f(x)`.
+
+        Parameters
+        ----------
+        x : N x D numpy.ndarray
+            Input RAS+ coordinates (i.e., physical coordinates).
+        inverse : bool
+            If ``True``, apply the inverse transform :math:`x = f^{-1}(y)`.
+
+        Returns
+        -------
+        y : N x D numpy.ndarray
+            Transformed (mapped) RAS+ coordinates (i.e., physical coordinates).
+
+        Examples
+        --------
+        >>> xfm = LinearTransformsMapping([
+        ...     [[1., 0, 0, 1.], [0, 1., 0, 2.], [0, 0, 1., 3.], [0, 0, 0, 1.]],
+        ...     [[1., 0, 0, -1.], [0, 1., 0, -2.], [0, 0, 1., -3.], [0, 0, 0, 1.]],
+        ... ])
+        >>> xfm.matrix
+        array([[[ 1.,  0.,  0.,  1.],
+                [ 0.,  1.,  0.,  2.],
+                [ 0.,  0.,  1.,  3.],
+                [ 0.,  0.,  0.,  1.]],
+        <BLANKLINE>
+               [[ 1.,  0.,  0., -1.],
+                [ 0.,  1.,  0., -2.],
+                [ 0.,  0.,  1., -3.],
+                [ 0.,  0.,  0.,  1.]]])
+
+        >>> y = xfm.map([(0, 0, 0), (-1, -1, -1), (1, 1, 1)])
+        >>> y[0, :, :3]
+        array([[1., 2., 3.],
+               [0., 1., 2.],
+               [2., 3., 4.]])
+
+        >>> y = xfm.map([(0, 0, 0), (-1, -1, -1), (1, 1, 1)], inverse=True)
+        >>> y[0, :, :3]
+        array([[-1., -2., -3.],
+               [-2., -3., -4.],
+               [ 0., -1., -2.]])
+
+
+        """
+        affine = self.matrix
+        coords = _as_homogeneous(x, dim=affine.shape[-1] - 1).T
+        if inverse is True:
+            affine = np.linalg.inv(affine)
+        return np.swapaxes(affine.dot(coords), 1, 2)
+
+    def to_filename(self, filename, fmt='X5', moving=None):
+        """Store the transform in BIDS-Transforms HDF5 file format (.x5)."""
+        if fmt.lower() in ('itk', 'ants', 'elastix'):
+            itkobj = io.itk.ITKLinearTransformArray.from_ras(self.matrix)
+            itkobj.to_filename(filename)
+            return filename
+
+        # Rest of the formats peek into moving and reference image grids
+        if moving is not None:
+            moving = ImageGrid(moving)
+        else:
+            moving = self.reference
+
+        if fmt.lower() == 'afni':
+            afniobj = io.afni.AFNILinearTransformArray.from_ras(
+                self.matrix, moving=moving, reference=self.reference)
+            afniobj.to_filename(filename)
+            return filename
+
+        if fmt.lower() == 'fsl':
+            fslobj = io.fsl.FSLLinearTransformArray.from_ras(
+                self.matrix, moving=moving, reference=self.reference
+            )
+            fslobj.to_filename(filename)
+            return filename
+
+        if fmt.lower() in ('fs', 'lta'):
+            # xform info
+            # to make LTA file format
+            lta = io.LinearTransformArray()
```

Comment:
should we rename this to `FSLinearTransformArray` to be uniform with the other LTAs?

Reply:
I think we can keep lta. But, if we were to change it, let's in a different pr